doc/getting_started/index.rst

```diff
@@ -96,6 +96,17 @@ Clone all of Zephyr's repositories in a new :file:`zephyrproject` directory:
 You can replace :file:`zephyrproject` with another directory name. West creates
 the directory if it doesn't exist. See :ref:`west-multi-repo` for more details.
 
+.. important::
+
+   You need to run ``west update`` whenever you pull the :file:`zephyr`
```

Comment:
You're describing the issue in terms of its implementation (yml file changes) and not what the user is doing to cause it, so how about this instead:
```
   You need to run ``west update`` any time you pull the :file:`zephyr`
   repository or switch branches.  This will keep :ref:`modules` in
   the :file:`zephyrproject` folder in sync with the zephyr repository.
```

Reply:
Saying "switch branches" is not quite right -- the current commit is not always on a branch; you could also have a tag checked out, or just some random detached HEAD. You also need to run west.yml if you update its contents yourself, which changes the working tree but not the current HEAD.

The TL;DR common denominator is that anytime you change west.yml, you should run west update.

This is also true of every step in a `git bisect` -- it changes the current commit, which means you need to run west update, as bisecting changes the current working tree even though you haven't explicitly run `git checkout`.

Reply:
```
   You need to run ``west update`` any time :file:`zephyr/west.yml` changes.
   This command keeps :ref:`modules` in the :file:`zephyrproject` folder in sync
   with the code in the zephyr repository, so they work correctly together.

   Some examples when ``west update`` is needed are: whenever you
   pull the :file:`zephyr` repository, switch branches in it, or perform a ``git
   bisect`` inside of it.
```

@dbkinder how is that?

Reply:
Works for me.  

I've got an update to the Getting Started Guide in the works (waiting for the 2.0 release), based on the recent developer experience review at the TSC, that's a stripped-down guide for a new user's first try with getting blinky built and running.  We'll be moving all the complicated stuff to the developer guide where the exceptions, alternatives, and complications can be explained without scaring the "new user" looking at developing with Zephyr for the first time.  (That's what @nashif mentions.)

Reply:
> Works for me.

Thank you, captain! I've made it so :).

> I've got an update to the Getting Started Guide in the works (waiting for the 2.0 release), 

I heard about and am really looking forward to that.

Did your study indicate that people have trouble with environment variables?

I would like to remove the need for zephyr-env.sh in the west based workflow. This is relevant to your work here. Maybe we can discuss in another forum later.